CNV/cnv-2.1.0.sh

```diff
@@ -154,3 +154,68 @@ metadata:
   namespace: "${TARGET_NAMESPACE}"
 spec: {}
 EOF
+
+echo "Waiting for HCO to get fully deployed"
+oc wait -n ${TARGET_NAMESPACE} hyperconverged hyperconverged-cluster --for condition=Available --timeout=10m
+
+echo "Configuring network on nodes"
+
+echo "Detecting the default interface"
+machine_cidr=$(grep 'machineCIDR' ../OpenShift/install-config.yaml | sed 's/\(.*\): *\(.*\)/\2/')
+while ! default_iface=$(oc get nodenetworkstate ${node} -o jsonpath="{.items[0].status.currentState.routes.running[?(@.destination==\"${machine_cidr}\")].next-hop-interface}" | cut -d " " -f 1); do
+    sleep 10
+done
+
+if [ "${default_iface}" == "brext" ]; then
+    echo "Bridge brext seems to be already configured as the default interface, skipping the rest of network setup"
+    exit 0
+fi
+
+echo "Detecting the secondary interface"
+management_cidr=172.22.0.0/24
+while ! secondary_iface=$(oc get nodenetworkstate ${node} -o jsonpath="{.items[0].status.currentState.routes.running[?(@.destination==\"${management_cidr}\")].next-hop-interface}" | cut -d " " -f 1); do
+    sleep 10
+done
+
+echo "Applying node network configuration policy"
+cat <<EOF | oc apply -f -
+apiVersion: nmstate.io/v1alpha1
+kind: NodeNetworkConfigurationPolicy
+metadata:
+  name: kni-policy
+spec:
+  desiredState:
+    interfaces:
+    - name: brext
+      type: linux-bridge
+      state: up
+      ipv4:
+        dhcp: true
+        enabled: true
+      ipv6:
+        dhcp: true
+        enabled: true
+      bridge:
+        options:
+          stp:
+            enabled: false
+        port:
+        - name: ${default_iface}
+    - name: ${secondary_iface}.3435
```

Comment:
begs to defined STORAGE_VLAN=3435 as it appears twice.

Reply:
Also, do we want to place this in cnv.sh? This is required for ocs, not cnv. Is it a terrible overhead to create another policy under OCS?

Reply:
> Also, do we want to place this in cnv.sh? This is required for ocs, not cnv. Is it a terrible overhead to create another policy under OCS?

The issue is that we need nmstate installed by HCO and that happens in this script. Should we assume the order in which are those two executed?

Reply:
Only now I see that this script is a copy of another: https://github.com/openshift-kni/install-scripts/tree/master/CNV#cnv-21-for-htb. I will split this code outside the `cnv-2.1.0.sh`. Any suggestions on where should I put this network script? Please note that it needs CNV to be installed first. @sreichar

Reply:
> > Also, do we want to place this in cnv.sh? This is required for ocs, not cnv. Is it a terrible overhead to create another policy under OCS?
> 
> The issue is that we need nmstate installed by HCO and that happens in this script. Should we assume the order in which are those two executed?

@phoracek, since you suggest that CNV would br the thing that creates OCS's vlan, we have to make sure CNV is installed first.

I would appreciate if we leave OCS out of this PR. Linux bridge is complex enough, and time is short enough, and OCS can work on the native VLAN. Let us focus on the bridge, have it merged, and then spend time to optimize OCS.

Reply:
Please correct me if I'm wrong here @sreichar, but I believe the storage VLAN has to be configured. What I don't know is whether the order matters. Is it an issue if we configure the VLAN only after the OCS is installed?

We still don't have keepalived fix available (published, merged), so we can't merge this PR now. On top of that, the VLAN configuration seems to work just fine.

@dankenigsberg in case the order of configuration doesn't matter and VLAN is behaving fine, I'd like to keep those two together. If there is anything fishy with VLAN once we get keepalived, I will split this. Is that fine?